src/app/Plugins/Weather/Components/weather/weather.component.ts

```diff
@@ -21,7 +22,9 @@ export class WeatherComponent implements OnInit {
   locations: Location[];
   selectedLocation: string;
 
-  constructor(private weatherService: WeatherDataService) { }
+  constructor(private readonly databaseService: BackendResolver, private readonly weatherService: WeatherDataService) {
+    this.weatherService.path = databaseService.backendPath;
```

Comment:
It would be better if the WeatherDataService got a BackendResolver injected and grabbed the server host from that, not the component using it